### PR TITLE
Refactor only: separate subscriptions from ethereumclient service

### PR DESCRIPTION
**Pull request by KillariDev**

### CODE REVIEW

File: extension/app/ts/simulation/services/ETHSubscriptionService.ts [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+import { ErrorWithData } from '../../utils/errors.js'
+import { Future } from '../../utils/future.js'
+
+type Subscription = {
+	callback: (subscriptionId: string, reply: JsonRpcNewHeadsNotification) => void
+	params: EthSubscribeParams,
+	rpcSocket: WebSocket
+}
+
+export class ETHSubscriptionService {
```
Comment (MicahZoltu) [inline extension/app/ts/simulation/services/ETHSubscriptionService.ts:11]: To match naming convention elsewhere:
```suggestion
export class EthereumSubscriptionService {
```
Alternatively (not a fan of this but better than `ETH`):
```suggestion
export class EthSubscriptionService {
```

File: extension/app/ts/simulation/services/ETHSubscriptionService.ts [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+	}
+
+	public readonly createSubscription = async (params: EthSubscribeParams, callback: (subscriptionId: string, reply: JsonRpcNewHeadsNotification) => void) => {
+		switch(params.params[0]) {
+			case 'newHeads': {
+				const rpcSocket = new WebSocket(this.webSocketConnectionString)
+				const subscriptionId = new Future<string>()
+
+				rpcSocket.addEventListener('open', _event => {
+					const request = { jsonrpc: '2.0', id: 0, method: 'eth_subscribe', params: ['newHeads'] }
```
Comment (MicahZoltu) [inline extension/app/ts/simulation/services/ETHSubscriptionService.ts:37]: Out of scope of this PR, but I recommend against using `eth_subscribe` and instead just polling every block.

**Review by MicahZoltu — APPROVED** [review]

File: extension/app/ts/simulation/services/ETHSubscriptionService.ts (hunk 1 above)
Comment (KillariDev) [reply extension/app/ts/simulation/services/ETHSubscriptionService.ts:11]: the point was to refer to `eth_subscription` with the naming

File: extension/app/ts/simulation/services/ETHSubscriptionService.ts (hunk 2 above)
Comment (KillariDev) [reply extension/app/ts/simulation/services/ETHSubscriptionService.ts:37]: yeah, there's PR for it: https://github.com/DarkFlorist/TheInterceptor/issues/302

File: extension/app/ts/simulation/services/ETHSubscriptionService.ts (hunk 1 above)
Comment (MicahZoltu) [reply extension/app/ts/simulation/services/ETHSubscriptionService.ts:11]: I recommend not actually implementing this service using `eth_subscription` under the hood, hence I recommend not porting that name over to the class.  😀

**Pull request merged**